src/models/apps/FundsRescuer.js

```diff
@@ -0,0 +1,100 @@
+    logger.info(`Upgrading fundraising Token Manager to ${newTokenManagerBase}`)
+
+    // Grant "CLAIM_TOKENS_ROLE" permission on fundraising Token Manager to governor's Voting
+    const createClaimTokensRoleData = this.encoder.encodeCreatePermission(governor.voting, fundraisingTokenManagerAddress, sha3('CLAIM_TOKENS_ROLE'), governor.voting)
+    const createClaimTokensRoleExecuteData = this.encoder.encodeExecute(fundraisingACL, 0, createClaimTokensRoleData)
+    callsScript.push({ to: governor.agent, data: createClaimTokensRoleExecuteData })
+    logger.info(`Granting CLAIM_TOKENS_ROLE on fundraising Token Manager to governor's Voting ${governor.voting}`)
+
+    // Claim and transfer DAI tokens
+    const claimTokensData = this.encoder.encodeClaimTokens(dai.address, dai.recipient)
+    callsScript.push({ to: fundraisingTokenManagerAddress, data: claimTokensData })
+    logger.info(`Claiming DAI for ${dai.recipient}`)
+
+    // Revoke "CLAIM_TOKENS_ROLE" permission on fundraising Token Manager to governor's Voting
+    const revokeClaimTokensRoleData = this.encoder.encodeRevokePermission(governor.voting, fundraisingTokenManagerAddress, sha3('CLAIM_TOKENS_ROLE'))
+    callsScript.push({ to: fundraisingACL, data: revokeClaimTokensRoleData })
+    logger.info(`Revoking CLAIM_TOKENS_ROLE on fundraising Token Manager from governor's Voting ${governor.voting}`)
+
+    // Remove governor's Voting as the permission manager of "CLAIM_TOKENS_ROLE" on fundraising Token Manager
+    const removeClaimTokensPermissionManagerData = this.encoder.encodeRemovePermissionManager(fundraisingTokenManagerAddress, sha3('CLAIM_TOKENS_ROLE'))
+    callsScript.push({ to: fundraisingACL, data: removeClaimTokensPermissionManagerData })
+    logger.info(`Removing governor's Voting ${governor.voting} as the permission manager of CLAIM_TOKENS_ROLE on fundraising Token Manager`)
+
+    // Downgrade Token Manager app of the fundraising DAO
+    const downgradeTokenManagerData = await this.encoder.encodeSetApp(BASE_NAMESPACE, fundraisingTokenManagerAppId, currentFundraisingTokenManagerBase)
+    const downgradeTokenManagerExecuteData = this.encoder.encodeExecute(fundraisingKernelAddress, 0, downgradeTokenManagerData)
+    callsScript.push({ to: governor.agent, data: downgradeTokenManagerExecuteData })
+    logger.info(`Downgrading fundraising Token Manager version to base ${currentFundraisingTokenManagerBase}`)
+
+    // Recover ANJ tokens from old registry
+    const recoverAnjFundsData = this.encoder.encodeRecoverFunds(anj.address, governor.agent)
+    const recoverAnjFundsExecuteData = this.encoder.encodeExecute(oldJurorsRegistry, 0, recoverAnjFundsData)
+    callsScript.push({ to: governor.agent, data: recoverAnjFundsExecuteData })
+    logger.info(`Recovering ANJ locked in old registry ${oldJurorsRegistry}`)
+
+    // Grant "TRANSFER_ROLE" permission on governor's Agent to governor's Voting
```

Comment:
I'm guessing we can't specify an alternative recipient for the governor, so we have to do this transfer dance?

Reply:
Exactly